chain/ethereum/src/adapter.rs

```diff
@@ -35,6 +35,53 @@ use crate::{Chain, Mapping, ENV_VARS};
 pub type EventSignature = H256;
 pub type FunctionSelector = [u8; 4];
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct EventSignatureWithTopics {
```

Comment:
I think this matches in a somewhat unexpected way (for me at least) so i think ti would be worth adding some rust docs here to explain what this is used for and how it works. Maybe add some illustrative examples